pkg/monitor/Dockerfile

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2024 Zededa, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# syntax=docker/dockerfile:1
+ARG RUST_VERSION=lfedge/eve-rust:1.80.1
+FROM --platform=$BUILDPLATFORM ${RUST_VERSION} AS toolchain-base
+ARG TARGETARCH
+
+FROM toolchain-base AS target-amd64
+ENV CARGO_BUILD_TARGET="x86_64-unknown-linux-musl"
+
+FROM toolchain-base AS target-arm64
+ENV CARGO_BUILD_TARGET="aarch64-unknown-linux-musl"
+
+FROM toolchain-base AS target-riscv64
+ENV CARGO_BUILD_TARGET="riscv64gc-unknown-linux-gnu"
+
+FROM target-$TARGETARCH AS toolchain
+RUN echo "Cargo target: $CARGO_BUILD_TARGET"
+
+FROM toolchain AS planer
+# get sources from GIT
+ADD --keep-git-dir=true https://github.com/lf-edge/eve-monitor-rs.git /app
+
+WORKDIR /app
+# create a recipe
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM toolchain AS cacher
+# copy the recipe
+WORKDIR /app
+COPY --from=planer /app/recipe.json recipe.json
+# build the dependencies
+RUN cargo chef cook --release --recipe-path recipe.json
+
+# building the final image
+FROM toolchain AS builder
+ADD --keep-git-dir=true https://github.com/lf-edge/eve-monitor-rs.git /app
```

Comment:
Did somebody review this code?

Reply:
Not the entire code, but there is at least one PR to be reviewed: https://github.com/lf-edge/eve-monitor-rs/pull/3